### ISSUE TITLE
Traverses leaves and finds multiple trees to harvest

### ISSUE THREAD

**Issue by lorilith** [Levoment/SuperAxes#1]
This mod seems to harvest trees attached by leaves alone. for example, if i have an oak tree and a birch tree grown a few blocks away, but their leaves are touching, the superaxe harvests both.

it seems that while traversing the tree, leaves are added to the traversal list causing any attached leaves to attach logs. 

best i can come up with is only processing the block if lastpos is either a log or the the current checked block is a leaf. this would unfortunately leave diagonal log blocks alone but, if breaking leaves is enabled, would break leaves of all connected trees too.

**Comment by Levoment** [comment]
Yes. That is correct. I noticed the behaviour as well. I left a comment on the [pull request](https://github.com/Levoment/SuperAxes/pull/2).

I've been trying to come up with a solution for that, but as of now, nothing comes to mind. The solutions I've stumbled upon would make the SuperAxes unable to harvest leaves if implemented.

I'll continue working on finding a solution for it and leave it open in case someone finds a solution.
